src/pybind/mgr/dashboard/services/rbd.py

```diff
@@ -630,8 +631,14 @@ def delete(cls, image_spec):
 
         image = RbdService.get_image(image_spec)
         snapshots = image['snapshots']
-        for snap in snapshots:
-            RbdSnapshotService.remove_snapshot(image_spec, snap['name'], snap['is_protected'])
+        if len(snapshots) > 0:
+            for snap in snapshots:
+                RbdSnapshotService.remove_snapshot(image_spec, snap['name'], snap['is_protected'])
+            while True:
```

Comment:
Is it possible to have a timeout here? I don't know how I feel having an "infinite" loop.

Reply:
hi, thanks for your review. I have referred to multiple codes of infinite loop，such as 'delete osd'.That's what I think, if ceph cluster is healthy, the loop will definitely end;conversely the priority of this infinite loop is very low.